runner.go

```diff
@@ -202,61 +202,78 @@ func filterExecuteTests(tests []*rainforest.RFTest, tags []string, forceExecute,
 }
 
 func monitorRunStatus(c cliContext, runID int) error {
-	// Create two channels to communicate with the polling goroutine
-	// One that will tick when it's time to poll and the other to gather final state
-	t := time.NewTicker(runStatusPollInterval)
-	statusChan := make(chan statusWithError, 0)
-	go updateRunStatus(c, runID, t, statusChan)
+	backoff := 1
 
-	// This channel readout will block until updateRunStatus pushed final result to it
-	finalState := <-statusChan
-	if finalState.err != nil {
-		return cli.NewExitError(finalState.err.Error(), 1)
-	}
+	for {
+		status, msg, done, err := getRunStatus(c.Bool("fail-fast"), runID)
+		log.Print(msg)
 
-	if finalState.status.FrontendURL != "" {
-		log.Printf("The detailed results are available at %v\n", finalState.status.FrontendURL)
-	}
+		if done {
+			if status.FrontendURL != "" {
+				log.Printf("The detailed results are available at %v\n", status.FrontendURL)
+			}
 
-	postRunJUnitReport(c, runID)
+			postRunJUnitReport(c, runID)
 
-	if finalState.status.Result != "passed" {
-		return cli.NewExitError("", 1)
-	}
+			if status.Result != "passed" {
+				return cli.NewExitError("", 1)
+			}
+
+			return nil
+		}
+
+		// If we hit an error, wait longer before retrying
+		if err != nil {
+			backoff++
```

Comment:
We should print something out saying that we are going to pause briefly so that it's clear why we are waiting for potentially 2 minutes between requests.